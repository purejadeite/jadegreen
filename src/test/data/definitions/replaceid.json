{
	"id": "book",
	"sheets": [
		{
			"comment": "キー置換シート",
			"id": "keyreplace",
		    "name": "validation",
		    "cells": [
		        {"id": "fields", "row": 6, "endKeyId": "fieldName", "options": [
		                {"type": "replaceid", "keyId": "validationType", "map": {
		                	"length": {"arg0": "min", "arg1": "max"},
		                	"maxlength": {"arg0": "length"}
		                }},
		                {"type": "group", "keyId": "fieldName", "toId": "validators"}
		            ],
		            "columns": [
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "fieldName", "column": 3},
		                {"id": "validationType", "column": 4, "options": [{"type": "lower"}]},
		                {"id": "arg0", "column": 5, "options": [{"type": "tonull"}, {"type": "remove"}]},
		                {"id": "arg1", "column": 6, "options": [{"type": "tonull"}, {"type": "remove"}]},
		                {"id": "arg2", "column": 7, "options": [{"type": "tonull"}, {"type": "remove"}]},
		                {"id": "arg3", "column": 8, "options": [{"type": "tonull"}, {"type": "remove"}]}
		            ]
		        }
		    ]
		}
	]
}