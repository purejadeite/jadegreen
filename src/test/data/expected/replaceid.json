[
	{
		"sheetName": "validation",
		"fields": [
			{
				"fieldName": "fileExistFlag",
				"validators": [
		   			{
						"logicalFieldName": "ファイル有無フラグ",
						"fieldName": "fileExistFlag",
						"validationType": "notnull"
					}
				]
			},
			{
				"fieldName": "fileName",
				"validators": [
		   			{
		   				"logicalFieldName": "ファイル名",
						"fieldName": "fileName",
						"validationType": "notnull"
					}
				]
			},
			{
				"fieldName": "filePath",
				"validators": [
					{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "notnull"
					},
		   			{
						"logicalFieldName": "ファイルパス",
						"fieldName": "filePath",
						"validationType": "length",
						"arg0": "2",
						"arg1": "10"
					}
				]
			},
			{
				"fieldName": "lines",
				"validators": [
		   			{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "notnull"
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "maxlength",
						"arg0": "100"
					},
					{
						"logicalFieldName": "行情報",
						"fieldName": "lines",
						"validationType": "kana"
					}
				]
			}
		]
	}
]